dcos/cosmospackage.py

```diff
@@ -289,9 +296,50 @@ def package_versions(self):
 
         params = {"packageName": self.name(), "packageVersions": True}
         url = urllib.parse.urljoin(self._cosmos_url, 'v1/package/describe')
-        response = http.post(url, json=params)
+        response = http.post(url, json=params,
+                             headers=_get_cosmos_header("describe"))
 
-        if response.status_code != 200:
-            raise DCOSException(response.json().get("message"))
+        if _cosmos_error(response):
+            raise DCOSException(response.json().get("errors"))
 
         return list(response.json().keys())
+
+
+def _get_header(request_type):
+    """Returns header str for talking with cosmos
+
+    :param request_type: name of specified request (ie uninstall-request)
+    :type request_type: str
+    :returns: header information
+    :rtype: str
+    """
+
+    return ("application/vnd.dcos.cosmos.{}+json;"
+            "charset=utf-8;version=v1").format(request_type)
+
+
+def _get_cosmos_header(request_name):
+    """Returns header fields needed for a valid request to cosmos
+
+    :param request_name: name of specified request (ie uninstall)
+    :type request_name: str
+    :returns: dict of required headers
+    :rtype: {}
+    """
+
+    return {"Accept": _get_header("{}-response".format(request_name)),
+            "Content-Type": _get_header("{}-request".format(request_name))}
+
+
+def _cosmos_error(response):
+    """Returns whether or not error object from cosmos
+
+    :param response: response from cosmos
+    :type response: Response
+    :returns: whether response has an error or not
+    :rtype: bool
+    """
+    content_type = response.headers.get('Content-Type')
+    if content_type is None:
+        return True
+    return _get_header("error") in content_type
```

Comment:
If we change this method to something like `_check_for_cosmos_error` and make it throw a `DCOSException` the we can change the `if ... : raise ...` pattern to just `_check_for_cosmos_error`. Even better we can use Python's decorators: http://simeonfranklin.com/blog/2012/jul/1/python-decorators-in-12-steps/

We have an example at: https://github.com/mesosphere/dcos-cli/blob/master/dcos/util.py#L533-L553